### PR TITLE
Add return Value type annotation

### PR DESCRIPTION
**Pull request by swamper123**

### CODE REVIEW

File: snap7/client.py [hunk 1]
```diff
@@ -31,25 +32,25 @@ class Client:
     A snap7 client
     """
 
-    def __init__(self):
+    def __init__(self) -> None:
```
Comment (gijzelaerr) [inline snap7/client.py:35]: I think `None` is the default return type, so you can save yourself the time to add it :)

File: snap7/client.py [hunk 2]
```diff
@@ -209,7 +210,7 @@ def upload(self, block_num):
         return bytearray(_buffer)
 
     @error_wrap
-    def download(self, data, block_num=-1):
+    def download(self, data, block_num=-1) -> Any:
```
Comment (gijzelaerr) [inline snap7/client.py:213]: Why do you use Any here?

File: snap7/common.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ def check_error(code, context="client"):
         raise Snap7Exception(error)
 
 
-def error_text(error, context="client"):
+def error_text(error, context="client") -> Array:
```
Comment (gijzelaerr) [inline snap7/common.py:69]: Array? why not just a str

File: snap7/logo.py [hunk 4]
```diff
@@ -53,7 +54,7 @@ def destroy(self):
         logger.info("destroying snap7 client")
         return self.library.Cli_Destroy(byref(self.pointer))
 
-    def disconnect(self):
+    def disconnect(self) -> Any:
```
Comment (gijzelaerr) [inline snap7/logo.py:57]: We should not use Any i think.

File: snap7/server.py [hunk 5]
```diff
@@ -42,10 +43,10 @@ def __init__(self, log=True):
         if log:
             self._set_log_callback()
 
-    def __del__(self):
+    def __del__(self) -> None:
         self.destroy()
 
-    def event_text(self, event):
+    def event_text(self, event) -> ctypes.Array:
```
Comment (gijzelaerr) [inline snap7/server.py:49]: why not jus tstr?

**Review by gijzelaerr — CHANGES_REQUESTED** [review]

**Comment by swamper123** [comment]
@gijzelaerr Don't spend to much time making reviews at the moment, since there will be changes and "tests" be pushed. 
When this Draft is ready, I will turn it into a normal PR and then you can take it apart. 🙈

**Comment by gijzelaerr** [comment]
I would only use `Any` if a function really can return an ambiguous type, or the number of types just becomes to complex to enumerate. But that is probably bad code anyway.

**Comment by swamper123** [comment]
Does somebody have a clue, why Mypy has a problem with `get_cpu_state(self) -> str:` ?
The value status_string is a string and can't get None, because it would be raised as an error then. 😕 

As well `load_library()` doesn't behave the way as intended. `self.library` is `None` and so the annotations are getting wrong, since an self.library.here_partner_func attribute seems to be expected.

**Comment by swamper123** [comment]
It seems to be a mypy problem itself.
There have been errors appearing when a function/method doesn't contain a `return None`. This could be solved by:` --no-warn-no-return` [source](https://mypy.readthedocs.io/en/stable/command_line.html?highlight=annotation%20return%20None#configuring-warnings)

How the situation looks like with the cdll data in mypy, I am still not sure.

**Comment by nikteliy** [comment]
There is some a problem with "Windows tests" after my last commit. But I have no idea how I could break it. The tests take too long.

**Comment by nikteliy** [comment]
@gijzelaerr  сould you cancel the test before github sues us for wasting computing resources?

**Comment by gijzelaerr** [comment]
at some point the test will just timeout, don't worry.

cancelled.

**Comment by swamper123** [comment]
I read through some PRs and Issues from mypy and it seems that 

> snap7/client.py:93: error: Incompatible types in assignment (expression has type "None", variable has type "str")

is caused by the redefinition from str to None in some cases. A Workaround may be `--allow-redefinition`.

The issues in partner.py is indeed a `load_library()` problem, which isn't loading the library (so that class attributes are available). 
@gijzelaerr Shall I undo the failing changes?

File: snap7/client.py [hunk 6]
```diff
@@ -49,55 +50,55 @@ def create(self):
...
         """
         Retrieves CPU state from client
         """
         state = c_int(0)
         self._library.Cli_GetPlcStatus(self._pointer, byref(state))
-
         try:
             status_string = cpu_statuses[state.value]
         except KeyError:
             status_string = None
```
Comment (gijzelaerr) [inline snap7/client.py:93]: i think mypy infers  `status_string` can be both a string and a None following this logic, if you just raise the snap exception in this block directly it probably won't be confused anymore.

Comment (swamper123) [reply snap7/client.py]: Jep, that worked. :)

**Comment by gijzelaerr** [comment]
there is now a conflict with your previous pr

**Comment by swamper123** [comment]
Fresh rebased

**Review by gijzelaerr — APPROVED** [review]

**Pull request merged**